rpm/resalloc.spec.tpl

```diff
@@ -33,6 +33,18 @@ the purposes of CI/CD tasks.
 %bcond_with    python3
 %endif
 
+# Modern distributions (using RPM v4.19+; for example, Fedora 39+) do not
+# require the %%pre scriptlet for creating users/groups because the sysusers
+# feature is now built directly into RPM.  Simply including the sysusers
+# `mock.conf` file in a package payload is sufficient to leverage this feature.
+# However, for older distributions that lack this capability, we still define
+# the %%pre scriptlet.
+%if (0%{?rhel} && 0%{?rhel} < 10) || (0%{?mageia} && 0%{?mageia} < 10) || (0%{?suse_version} && 0%{?suse_version} < 1660)
```

Comment:
Too bad there is no macro with RPM version because this is getting out of hand.

Reply:
Yeah. I just c&p'ed this from a place where people cared (mock), but I think we are just fine to support RHEL8+ and Fedora (if you think this is too ugly).

I think the problem is how to compare %rpmversion, without lua/bash heavy lifting...

Reply:
This is cool thing, btw. 
```
$ rpm --eval '%[v"4.20.0" > v"4.19.0"]'
1
```
But  this is RHEL9+ only: https://github.com/rpm-software-management/rpm/commit/145b5f87404e4c811b36bebd34e5602d08debc99

Reply:
```suggestion
%if %{?rpmversion:v"%{rpmversion}" >= v"4.19"}%{!?rpmversion:0}
```

If the RPM version is too old for the comparison and this feature, it does not have the macro defined.

Reply:
Oh, my condition is reversed. But you get the idea.

Reply:
And I already merged the PR. But good suggestion. Thank you.